src/tools/rbd/action/Kernel.cc

```diff
@@ -188,6 +221,10 @@ static int do_kernel_map(const char *poolname, const char *imgname,
 
   free(devnode);
 out:
+  if (r < 0) {
+    print_error_description(poolname, imgname, snapname, -r);
```

Comment:
We generally keep all the error codes negative, so just for consistency sake I would recommend keeping all error code comparisons to the negative version.